metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/interfaces/MessageParser.java

```diff
@@ -34,15 +34,15 @@
    * @param rawMessage
    * @return If null is returned, this is treated as an empty list.
    */
-  List<T> parse(byte[] rawMessage);
+  List<T> parse(byte[] rawMessage, SensorParserConfig sensorParserConfig);
```

Comment:
@nickwallen The differences between `configurationChanged` and `configure` is that `configure` has an existing intent and this would change it, we would need to ensure every parser's configure implementation is compatible with this change in intent and doesn't cause regression.

Regarding the similarity in names, I obviously don't think they're too similar (otherwise I'd not have suggested it ;).  I think `configurationChanged` sounds like a hook that is called when configurations change, which is exactly what it would be.  `configure` has an existing understanding, which is a one-time configuration.

I've expressed a desire to get rid of one of `init` or `configure`, but I don't really care which.